homeassistant/components/tado/__init__.py

```diff
@@ -203,18 +207,35 @@ def update_mobile_devices(self) -> None:
             _LOGGER.error("Unable to connect to Tado while updating mobile devices")
             return
 
+        if len(mobile_devices) == 0:
+            _LOGGER.warning(
+                "No linked mobile devices found for home ID %s", self.home_id
+            )
+            return
+
+        if "errors" in mobile_devices and len(mobile_devices["errors"]) > 0:
+            _LOGGER.error(
+                "Error for home ID %s while updating mobile devices: %s",
+                self.home_id,
+                mobile_devices["errors"],
+            )
+            return
+
         for mobile_device in mobile_devices:
             self.data["mobile_device"][mobile_device["id"]] = mobile_device
+            _LOGGER.debug(
+                "Dispatching update to %s mobile device: %s",
+                self.home_id,
+                mobile_device,
+            )
+            dispatcher_send(
```

Comment:
Don't send a dispatch for each mobile device. Just dispatch once when connector data is complete. The dispatch signal just needs to add the home id. This is the index that is missing and causing the problem.

Reply:
Remove this dispatch and update the dispatch below with the new signal instead.